daemon/exec.go

```diff
@@ -237,6 +238,12 @@ func (daemon *Daemon) ContainerExecStart(ctx context.Context, name string, stdin
 	p.Env = ec.Env
 	p.Cwd = ec.WorkingDir
 	p.Terminal = ec.Tty
+	if p.Terminal && ec.ConsoleSize != nil {
+		p.ConsoleSize = &specs.Box{
+			Height: ec.ConsoleSize[0],
+			Width:  ec.ConsoleSize[1],
+		}
+	}
```

Comment:
🤔 I'm thinking now; should the size be part of the `POST /exec/{name:.*}/start` endpoint? Or should it be on both `POST /containers/{name:.*}/exec` and  `POST /exec/{name:.*}/start` ?

Basically thinking (and I think we were discussing that at some point);

- there could be a time between _creating_ and _starting_ the exec
- possibly depending on the scenario, it may be important to have the correct size when _starting_ the exec.
- In some scenarios (possibly) no new size may be passed to that request, in which case we can use the size that's already set during `exec create` 👈 this is what's currently implemented in this PR
- In other scenarios, the `exec start` request _may_ pass a new size. I think we could facilitate that by also adding the size to the `exec start` endpoint; this will be "optional"; if no new size is passed, use what was already set, otherwise update the size when starting the `exec`

Reply:
Yeah I think it makes sense to also be a part of the `start` endpoint. Also the same could apply to the `container start`.

Reply:
Yup, looking at it, I think It's probably safe to set it for both `create` and `start` (and both for `container` and `exec`)

Reply:
Done